perception/autoware_tensorrt_bevdet/src/alignbev_plugin.cu

```diff
@@ -0,0 +1,352 @@
+/*
+ * Copyright (c) 2021-2023, NVIDIA CORPORATION. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "autoware/tensorrt_bevdet/alignbev_plugin.hpp"
+#include "autoware/tensorrt_bevdet/common.hpp"
+
+#include <cuda_fp16.h>
+#include <cuda_runtime_api.h>
+
+static inline __device__ bool within_bounds_2d(int h, int w, int H, int W)
+{
+  return h >= 0 && h < H && w >= 0 && w < W;
+}
+
+template <typename T1, typename T2>
+__global__ void align_bev_kernel(
+  const int nthreads, const T1 * input, const float * trans, T2 * output, TensorDesc output_desc)
```

Comment:
The `align_bev_kernel` function uses `T1` and `T2` as template parameters.  While flexible, it might be beneficial to add input validation within the function or the plugin to ensure type compatibility (e.g., prevent mixing half-precision and single-precision inputs in unexpected ways).  Consider adding comments explaining the acceptable type combinations and any potential performance implications.